### PR TITLE
ThemesList: Use localize()

### PR DESCRIPTION
**Pull request by ockham**
This is remotely related to the React 15 update, #5116.

The trouble with using a higher-order component here is that it adds another level so that shallow rendering will no longer work as before. (See https://github.com/airbnb/enzyme/issues/242 for Enzyme's take on this kind of issue.) My strategy here is to also export the non-wrapped `ThemesList`, and to pass identity as `translate` in the test. CCing @aduth and @gziolo for a review of that strategy, and suggestions for possible alternatives.

(The signup step ES6ification is mainly so that we don't have to `require( 'components/themes-list' ).default` there, which would be ugly.)

To test -- check that `/design` works as before.

### CODE REVIEW

File: client/components/themes-list/test/index.jsx [hunk 1]
```diff
@@ -46,7 +40,8 @@ describe( 'ThemesList', function() {
 			loading: false,
 			fetchNextPage: noop,
 			getButtonOptions: noop,
-			onScreenshotClick: noop
+			onScreenshotClick: noop,
+			translate: x => x // Mock translate()
```
Comment (gziolo) [inline client/components/themes-list/test/index.jsx]: I like to use lodash `identity`, but that works, too :)

**Comment by gziolo** [comment]
This is my preferred solution :) We have `localize` HOC covered with tests, so we can assume it works as expected. At the same time exporting component to take advantage of shallow rendering makes sense, because it has several benefits:
* it's faster
* it tests component in isolation
* you don't need to have DOM initialised in your tests

**Comment by gziolo** [comment]
Tests pass, /design works properly. I would wait for feedback from @aduth, if we has no objections :shipit:

**Comment by aduth** [comment]
>My strategy here is to also export the non-wrapped `ThemesList`, and to pass identity as `translate` in the test.

Yep, this is exactly what I'd recommend. Same strategy applies to components that we wrap in `connect`, if we want to test component without necessarily tying to Redux state (i.e. component should be written to accept simple props whether they be from Redux `connect` or manually specified).

Looks good 👍

**Pull request merged**